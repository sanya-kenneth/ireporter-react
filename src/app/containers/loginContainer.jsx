import React, { Component } from 'react';
import { connect } from 'react-redux';
import { loginAction } from '../../redux/actions/loginAction';
import LoginComponent from '../components/loginComponent.jsx';


export class LoginContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
    };
  }

    handleLogin = (event) => {
      event.preventDefault();
      const {
        email,
        password,
      } = this.state;
      const { loginAction } = this.props;

      loginAction(
        email,
        password,
      );
    }

    handleChange = (event) => {
      this.setState({ [event.target.id]: event.target.value });
    }

    render() {
      const { loginErrors, loginResponse } = this.props;
      return (
          <div>
              <LoginComponent
                  loginErrors={loginErrors}
                  loginResponse={loginResponse}
                  handleChange={this.handleChange}
                  handleLogin={this.handleLogin}
                />
            </div>
      );
    }
}

export const mapStateToProps = state => ({
  loginErrors: state.login.loginErrors,
  loginResponse: state.login.loginResponse,
});

export default connect(
  mapStateToProps,
  { loginAction },
)(LoginContainer);
