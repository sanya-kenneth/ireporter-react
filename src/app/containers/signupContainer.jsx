import React, { Component } from 'react';
import { connect } from 'react-redux';
import { signupAction } from '../../redux/actions/signupAction';
import SignupComponent from '../components/signupComponent.jsx';


export class SignupContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      firstname: '',
      lastname: '',
      othernames: '',
      phonenumber: '',
    };
  }

    handleSignup = (event) => {
      console.log('submit me please');
      event.preventDefault();
      const {
        username,
        email,
        password,
        firstname,
        lastname,
        phonenumber,
        othernames,
      } = this.state;
      const { signupAction } = this.props;

      signupAction(
        username,
        firstname,
        lastname,
        othernames,
        email,
        password,
        phonenumber,
      );
    }

    handleChange = (event) => {
      this.setState({ [event.target.id]: event.target.value });
    }

    render() {
      console.log(this.state);
      const { signupErrors, signupResponse } = this.props;
      return (
          <div>
              <SignupComponent
                  signupErrors={signupErrors}
                  signupResponse={signupResponse}
                  handleChange={this.handleChange}
                  handleSignup={this.handleSignup}
                />
            </div>
      );
    }
}

export const mapStateToProps = state => ({
  signupErrors: state.signup.signupErrors,
  signupResponse: state.signup.signupResponse,
});

export default connect(
  mapStateToProps,
  { signupAction },
)(SignupContainer);
