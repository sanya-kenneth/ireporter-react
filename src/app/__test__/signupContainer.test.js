import React from 'react';
import { shallow } from 'enzyme';
import { SignupContainer, mapStateToProps } from '../containers/signupContainer';

describe('<SignupContainer/>', () => {
  let wrapper;
  const props = {};
  beforeEach(() => {
    wrapper = shallow(<SignupContainer {...props} />);
  });
  it('should render SignupContainer without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should test the handleChange event', () => {
    const event = {
      target: {
        id: 'username',
        value: 'sanya',
      },
    };
    wrapper.instance().handleChange(event);
    expect(wrapper.instance().state.username).toBe('sanya');
  });
  it('should handle signUp event', () => {
    const props = {
      signupAction: jest.fn(),
    };
    const wrapper = shallow(<SignupContainer {...props} />);
    const instance = wrapper.instance();
    instance.handleSignup({ preventDefault: jest.fn() });
    expect(props.signupAction).toHaveBeenCalled();
  });
  it('should map state to props', () => {
    const initialState = {
      signup: {
        signupErrors: {},
        signupResponse: {
        },
      },
    };
    expect(mapStateToProps(initialState)).toEqual({
      signupErrors: {},
      signupResponse: {
      },
    });
  });
});
