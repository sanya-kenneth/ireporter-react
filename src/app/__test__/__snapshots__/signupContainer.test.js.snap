// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupContainer/> should render SignupContainer without crashing 1`] = `ShallowWrapper {}`;
